util/scripts/check-system.sh

```diff
@@ -13,14 +15,27 @@ print_no() {
     echo -e " → \e[1;31mNo\e[0m"
 }
 
+# Checks if $1 >= $2 with both parameters treated as version numbers.
+version_at_least() {
+    printf '%s\n%s\n' "$2" "$1" | sort --check=quiet --version-sort
+}
+
 
 echo "Checking tools required to build Tobira..."
 exit_code=0
 
 
 printf "▸ Rust installed? ('rustc' and 'cargo')"
 if has_command rustc && has_command cargo; then
-    print_yes
+    rust_version=$(rustc -V | sed -nr 's/rustc ([.0-9]+) .+/\1/p')
```

Comment:
I'm a big fan of long options in scripts. `man sed` also recommends using `-E` over `-r` for compatibility. But yeah, why not

```suggestion
    rust_version=$(rustc -V | sed --quiet --regexp-extended 's/rustc ([.0-9]+) .+/\1/p')
```